tests/system/CodeIgniterTest.php

```diff
@@ -360,6 +360,56 @@ public function testRunRedirectionWithURINotSet()
 		$this->assertEquals('http://example.com/pages/notset', $response->getHeader('Location')->getValue());
 	}
 
+	public function testRunRedirectionWithHTTPCode303()
+	{
+		$_SERVER['argv']            = [
+			'index.php',
+			'example',
+		];
+		$_SERVER['argc']            = 2;
+		$_SERVER['REQUEST_URI']     = '/example';
+		$_SERVER['SERVER_PROTOCOL'] = 'HTTP/1.1';
+		$_SERVER['REQUEST_METHOD']  = 'POST';
+
+		// Inject mock router.
+		$routes = Services::routes();
+		$routes->addRedirect('example', 'pages/notset', 301);
+
+		$router = Services::router($routes, Services::request());
+		Services::injectMock('router', $router);
+
+		ob_start();
+		$this->codeigniter->useSafeOutput(true)->run();
+		ob_get_clean();
+		$response = $this->getPrivateProperty($this->codeigniter, 'response');
+		$this->assertEquals('303', $response->getStatusCode());
```

Comment:
Why is this code changed to 303?
The route sets the status code 301.

Reply:
I sent a PR to fix the behavior: #7445

Reply:
I mean, it should be 301 ?

Reply:
Essentially it should be 301, but the handler is written so that the custom response code is ignored.